lib/vrl/parser/src/parser.lalrpop

```diff
@@ -244,13 +246,18 @@ Assignment: Node<Assignment> = {
 
 AssignmentSingle: Assignment = {
     <target: Sp<AssignmentTarget>> "="
-        <expr: Box<Expr>> => Assignment::Single { target, expr },
+        <expr: Box<Expr>> => Assignment::Single { target, op: AssignmentOp::Assign, expr },
+    <target: Sp<AssignmentTarget>> "|="
+        <expr: Box<Expr>> => Assignment::Single { target, op: AssignmentOp::Merge, expr },
```

Comment:
No need to do this here, but I think we should be able to merge these two and have a new `AssignmentOp` non-terminal that matches either on `=` and `|=` and then returns the correct enum variant.

Similar to how we have an `AssignmentTarget` non-terminal as well.

(same applies to the infallible assignment below)